hack/components/common_functions.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+set -xe
+
+function fetch_component() {
+    destination=$1
+    url=$2
+    commit=$3
```

Comment:
Is usually good practice to use `local` prefix so they don't get exposure out of the function.